### ISSUE TITLE
Feature Request: Add CheckpointSaverListener to tf.contrib.learn.Experiment

### ISSUE THREAD

**Issue by ankitvgupta** [tensorflow/tensorflow#16203]
Hello, 

I'm using the `tf.contrib.learn.Experiment` system to manage experiments built using the `tf.Estimator` framework. This setup allows me to specify checkpoint saving frequencies very easily, using just the `min_eval_frequency` argument to `tf.contrib.learn.Experiment`. However, I would like to be able to add a `tf.train.CheckpointSaverListener` (for example, to upload files to AWS after each checkpoint). Can there be a param added to `tf.contrib.learn.Experiment` to pass an optional `CheckpointSaverListener` object, which is then passed to the underlying `CheckpointSaverHook` object?

Thanks!

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Amazon Deep Learning AMI
- **TensorFlow version (use command below)**: 1.4
- **Python version**: 3.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: CUDA 8
- **GPU model and memory**: NVIDIA K80
- **Exact command to reproduce**: N/A

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by aselle** [comment]
@ispirmustafa, could you consider this feature request. Perhaps if it sounds good, it would make a great contributions welcome suggestion?

**Comment by ispirmustafa** [comment]
Agreed. Added to contributions.

**Comment by ankitvgupta** [comment]
Ok, I think I have a pretty simple PR that addresses this, and also adds early stopping params for the ValidationMonitor. It is here: https://github.com/tensorflow/tensorflow/pull/16391

**Comment by ankitvgupta** [comment]
Also, it looks like my feature request was already partly implemented in the master branch. However, the train_and_evaluate function was not using the parameter that was added to the constructor.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by ispirmustafa** [comment]
@xiejw this is about adding saver listeners to TrainSpec

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by ankitvgupta** [comment]
Closing this since Experiment is deprecated.
